modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/RemoveProcessor.java

```diff
@@ -75,8 +102,39 @@ public RemoveProcessor create(
             String description,
             Map<String, Object> config
         ) throws Exception {
+            final List<TemplateScript.Factory> compiledTemplatesToRemove = getTemplates(processorTag, config, "field");
+            final List<TemplateScript.Factory> compiledTemplatesToKeep = getTemplates(processorTag, config, "keep");
+
+            if (compiledTemplatesToRemove.isEmpty() && compiledTemplatesToKeep.isEmpty()) {
+                throw new IllegalArgumentException(
+                    "missing field [processors.remove.keep] or [processors.remove.field]. Please specify one of them."
+                );
+            }
+
+            if (compiledTemplatesToRemove.isEmpty() == false && compiledTemplatesToKeep.isEmpty() == false) {
+                throw new IllegalArgumentException(
+                    "Too many fields specified. Please specify either [processors.remove.keep] or [processors.remove.field]."
+                );
+            }
```

Comment:
For the sake of consistency, let's use the same exception and message style as other processors with mutually exclusive configuration options. See the [network direction processor](https://github.com/elastic/elasticsearch/blob/master/modules/ingest-common/src/main/java/org/elasticsearch/ingest/common/NetworkDirectionProcessor.java#L269) for an example.

Reply:
ok